librz/core/yank.c

```diff
@@ -94,40 +68,61 @@ static int perform_mapped_file_yank(RzCore *core, ut64 offset, ut64 len, const c
 	return res;
 }
 
-RZ_API int rz_core_yank_set(RzCore *core, ut64 addr, const ut8 *buf, ut32 len) {
-	// free (core->yank_buf);
-	if (buf && len) {
-		// FIXME: direct access to base should be avoided (use _sparse
-		// when you need buffer that starts at given addr)
-		rz_buf_set_bytes(core->yank_buf, buf, len);
-		core->yank_addr = addr;
-		return true;
-	}
-	return false;
+/* \brief Sets the contents of the yank clipboard
+ *
+ * Function sets the contents of the yank clibpoards as the raw bytes
+ *
+ * \p core RzCore instance
+ * \p addr Address "where" the information was yanked from
+ * \p buf Contents of the buffer to be set as the clipboard
+ * \p len Length of the buffer
+ */
+RZ_API bool rz_core_yank_set(RzCore *core, ut64 addr, const ut8 *buf, ut32 len) {
+	if (!buf || !len) {
+		return false;
+	}
+	// FIXME: direct access to base should be avoided (use _sparse
+	// when you need buffer that starts at given addr)
+	rz_buf_set_bytes(core->yank_buf, buf, len);
+	core->yank_addr = addr;
+	return true;
 }
 
-// Call set and then null terminate the bytes.
-RZ_API int rz_core_yank_set_str(RzCore *core, ut64 addr, const char *str, ut32 len) {
-	// free (core->yank_buf);
-	int res = rz_core_yank_set(core, addr, (ut8 *)str, len);
+/* \brief Sets the contents of the yank clipboard
+ *
+ * Function sets the contents of the yank clibpoards as the NULL-terminated string
+ *
+ * \p core RzCore instance
+ * \p addr Address "where" the information was yanked from
+ * \p str Zero-terminated string to be set as the clipboard
+ * \p len Length of the buffer
+ */
+RZ_API bool rz_core_yank_set_str(RzCore *core, ut64 addr, RZ_NONNULL const char *str, ut32 len) {
```

Comment:
```suggestion
RZ_API bool rz_core_yank_set_str(RzCore *core, ut64 addr, RZ_NONNULL const char *str) {
```
Then call `rz_core_yank_set` with `strlen(str)+1`